### ISSUE TITLE
Complete release of Version 2.1.11-2.1.14?

### ISSUE THREAD

**Issue by fingolfin** [simpcomp-team/simpcomp#27]
I see that you tried to release version 2.1.11 in July 2020. However, the release processes was never completed: there is neither a `v2.1.11` tag nor is this release visible on the website https://simpcomp-team.github.io/simpcomp/ -- and hence the GAP distribution never picked it up.

This looks a bit as if perhaps the tag was created locally but not pushed, and perhaps the website also was updated locally but not pushed?

That said, `CHANGELOG` states "First use of gap release tools". That's weird, as the release tools should at the very least guarantee that a tag is created, pushed to GitHub, and source files are uploaded there. It also update the website (though in the past, the default was that it left out the final "git push" for the website, it just printed a message telling you what to do; and an option could be used to tell it just do the push. I've since then flipped the default.

CC @alex-konovalov

**Comment by simpcomp-team** [comment]
The solution to this mystery is that when I tried to incorporate release tools I was interrupted by the beginning of the semester, and every since this got buried under a lot of other things. 

Last thing that happened was that the update failed because of generated files in my repo. I will now continue with the release by first removing those generated files...

**Comment by simpcomp-team** [comment]
... I assume some of the other issues about simpcomp relate to this problem.

**Comment by simpcomp-team** [comment]
Ok, so the release tools now did their job and the new release 2.1.12 should be complete.

**Comment by simpcomp-team** [comment]
Just changed it. I am currently making the complex library smaller so simpcomp uses less space. 

Once this is done I will make another release 2.1.13.

**Comment by simpcomp-team** [comment]
Just tried to release 2.1.13 and the release script fails to upload the release with command 

Pushing your branch to GitHub
remote: Repository not found.

I re-generated the access token as per README but it still does not work.

Also: I changed some details in "makedoc.g". It runs when I call it from within GAP, it runs when called for the first time by the release script, but it produces errors when it runs for the second time:


Extracting information from PackageInfo.g...
Package simpcomp 2.1.13
Building GAP package documentation (using makedoc.g)
Using archive basename simpcomp-2.1.13
Using GitHub repository gap-packages/simpcomp
Using GitHub username gap-packages
Using branch master
Creating git tag v2.1.13

Exporting repository content for tag 'v2.1.13'
Building GAP package documentation for archives (using makedoc.g)
ERROR: there was an error running GAP, see /home/jonathan/apps/gap-4.11.1/pkg/simpcomp/tmp/gap-error.log

With the content of gap-error.log:

Generating documentation in Directory("./doc/")
#I Composing XML document . . .
Error, Did not find chunk objecttypes at /home/jonathan/apps/gap-4.11.1/pkg/GAPDoc-1.6.4/lib/ComposeXML.gi:231 called from

**Comment by simpcomp-team** [comment]
Any idea what might be the problem?

**Comment by alex-konovalov** [comment]
> Pushing your branch to GitHub
remote: Repository not found.

In which directory do you run this?

**Comment by alex-konovalov** [comment]
> Using GitHub repository gap-packages/simpcomp

This looks like the source of the problem - you use own organisation, so should be `simpcomp-team/simpcomp`. This is taken from your `PackageInfo.g` file which has to be fixed - or you're welcome to migrate the repository to the `gap-packages` organisation, preserving all your admin rights for it.

**Comment by simpcomp-team** [comment]
Mhh... I thought I migrated this years ago... Apparently not. Thanks.

**Comment by jspreer** [comment]
Changed PackageInfo.g and now I am getting:

Extracting information from PackageInfo.g...
Package simpcomp 2.1.13
Building GAP package documentation (using makedoc.g)
Using archive basename simpcomp-2.1.13
Using GitHub repository simpcomp-team/simpcomp
Using GitHub username simpcomp-team
Using branch master
Using git tag v2.1.13 (derived from ArchiveURL in in PackageInfo.g)

Exporting repository content for tag 'v2.1.13'
Building GAP package documentation for archives (using makedoc.g)

**Comment by alex-konovalov** [comment]
The error message is not shown above but I've seen it elsewhere. 

I think 

https://github.com/simpcomp-team/simpcomp/blob/6d1becfb2da26704e7eebe1c0ccd9b50725f9cc7/makedoc.g#L5

need to be modified to add the `lib` directory where that chunk of the documentation is. I am not using AutoDoc for such setting, but hopefully 

    AutoDoc("simpcomp",rec(gapdoc:=rec(scan_dirs:=["doc/gapdoc","lib"])));

will just work. If not, more complex example of using AutoDoc on top of GAPDoc is https://github.com/gap-packages/recog/blob/master/makedoc.g.

You don't have to run a release to see if this is fixed: `gap makedoc.g` will attempt to build the manual, and you will see immediately if it worked or not.

**Comment by fingolfin** [comment]
AutoDoc always scans the `lib` dir by default.

sorry, can't help right now, am on vacation and without laptop :-). Can look again next week

**Comment by jspreer** [comment]
First of all, no reason to respond to this while on holiday. Enjoy your holiday. ;)

I understand what you are saying. The only reason I posted this under "ReleaseTools" is that running "makedoc.g" in gap always works. It even works the first time when called in "ReleaseTools". But then the second time it fails and is citing a file in the GAPDoc directory.

So, I am having trouble recreating the error in a setting I can control.

**Comment by alex-konovalov** [comment]
As far as the other issue with the full error message says,
```
Generating documentation in Directory("./doc/")
#I Composing XML document . . .
Error, Did not find chunk objecttypes at /home/jonathan/apps/gap-4.11.1/pkg/GAPDoc-1.6.4/lib/ComposeXML.gi:231 called from
```
and the chunk objecttypes is indeed there... (https://github.com/simpcomp-team/simpcomp/search?q=objecttypes)

**Comment by alex-konovalov** [comment]
Well, adding `lib` to 
```
AutoDoc("simpcomp",rec(gapdoc:=rec(scan_dirs:=["doc/gapdoc","lib"])));
```
fixed build for me. Still shows problems with multiple labels and other issues though:
```
Generating documentation in Directory("/Users/alexk/gap/gap-4.11.1/pkg/simpcomp/doc/")
#I Composing XML document . . .
#I Parsing XML document . . .
#I Checking XML structure . . .
#I Text version (also produces labels for hyperlinks):
#I First run, collecting cross references, index, toc, bib and so on . . .
#I Table of contents complete.
#I Producing the index . . .
#I Reading bibliography data files . . . 
#I   BibTeX format: biblio
#I   checking and translating to BibXMLext . . .
#I Writing bibliography . . .
#I Second run through document . . .
#I Producing simplified search strings and labels for hyperlinks . . .
#I Constructing LaTeX version and calling pdflatex:
#I Writing LaTeX file, 4 x pdflatex with bibtex and makeindex, 
#W There were LaTeX errors:
! Missing $ inserted.
<inserted text> 
                $
____________________
! Display math should end with $$.
<to be read again> 
                   \par 
____________________
! Missing $ inserted.
<inserted text> 
                $
____________________
! Missing $ inserted.
<inserted text> 
                $
____________________
! Missing { inserted.
<to be read again> 
                   ##
____________________
! You can't use `macro parameter character #' in math mode.
<recently read> ##
                  
____________________
! Missing } inserted.
<inserted text> 
                }
____________________
! Undefined control sequence.
<recently read> \scr 
                     
____________________
! Extra }, or forgotten $.
<recently read> }
                 
____________________
! Missing } inserted.
<inserted text> 
                }
____________________

#W There were LaTeX Warnings:
LaTeX Warning: Label `SCFacets' multiply defined.


____________________
LaTeX Warning: Label `SCFacetsEx' multiply defined.


____________________
LaTeX Warning: Label `SCCopy' multiply defined.


____________________
LaTeX Warning: Label `SCConnectedComponents' multiply defined.


____________________
LaTeX Warning: Label `SCDim' multiply defined.


____________________
LaTeX Warning: Label `SCEulerCharacteristic' multiply defined.


____________________
LaTeX Warning: Label `SCFVector' multiply defined.


____________________
LaTeX Warning: Label `SCFaceLattice' multiply defined.


____________________
LaTeX Warning: Label `SCFaceLatticeEx' multiply defined.


____________________
LaTeX Warning: Label `SCFpBettiNumbers' multiply defined.


____________________
LaTeX Warning: Label `SCIsConnected' multiply defined.


____________________
LaTeX Warning: Label `SCIsEmpty' multiply defined.


____________________
LaTeX Warning: Label `SCIsOrientable' multiply defined.


____________________
LaTeX Warning: Label `SCSkel' multiply defined.


____________________
LaTeX Warning: Label `SCSkelEx' multiply defined.


____________________
LaTeX Warning: Label `SCUnion' multiply defined.


____________________
LaTeX Warning: Label `SCHomology' multiply defined.


____________________
LaTeX Warning: Label `SCExportPolymake' multiply defined.

)
____________________
LaTeX Warning: Font shape declaration has incorrect series value `mc'.
               It should not contain an `m'! Please correct it.
               Found on input line 20.
____________________
LaTeX Warning: Font shape declaration has incorrect series value `mc'.
               It should not contain an `m'! Please correct it.
               Found on input line 23.
____________________
LaTeX Warning: Font shape declaration has incorrect series value `mc'.
               It should not contain an `m'! Please correct it.
               Found on input line 26.
____________________
LaTeX Warning: Font shape declaration has incorrect series value `mc'.
               It should not contain an `m'! Please correct it.
               Found on input line 46.
____________________
LaTeX Warning: There were multiply-defined labels.

Package atveryend Info: Empty hook `AtVeryVeryEnd' on input line 12224.
____________________

#W There are overfull boxes:
Overfull \hbox (29.58942pt too wide) in paragraph at lines 7894--7895
Overfull \hbox (12.6143pt too wide) in paragraph at lines 8048--8049
Overfull \hbox (29.58942pt too wide) in paragraph at lines 8269--8270
Overfull \vbox (26.40309pt too high) has occurred while \output is active []

#W BibTeX had warnings:
Warning--there's a number but no volume in Schulz94PolyhMnfPoly
#I Writing manual.six file ... 
#I Finally the HTML version . . .
#I First run, collecting cross references, index, toc, bib and so on . . .
#I Table of contents complete.
#I Producing the index . . .
#I Writing bibliography . . .
#I Second run through document . . .
#I - also HTML version for MathJax . . .
#I First run, collecting cross references, index, toc, bib and so on . . .
#I Table of contents complete.
#I Producing the index . . .
#I Writing bibliography . . .
#I Second run through document . . .
#I  File: /Users/alexk/gap/gap-4.11.1/pkg/simpcomp/doc/manual.lab written.
gap> 
```

**Comment by fingolfin** [comment]
If you merge PR #31 then a release should work....

**Comment by simpcomp-team** [comment]
Thanks for this. Much appreciated. I just made a release which looks mostly fine (2.1.13, originally planned a long long time ago - so sorry for not making the time to prioritising this earlier). 

There is a good chance that this still needs some work.

One thing I know for sure is that it does not pick up the correct chunks for the examples in the manual. This is something I still need to work on. Very likely related to 

`AutoDoc("simpcomp",rec(gapdoc:=rec(scan_dirs:=["doc/gapdoc","lib"])));`

which fixed things by taking the code from `lib` rather from `doc/gapdoc`... 

Can you give me some pointers on what the script is looking for in `scan_dirs` and why it rejects the ones in `doc/gapdoc`? Are there certain characters (such as < or >) the program has issues with? The `doc/gapdoc` files have lots of `&lt;` and `&gt;` in it and I remember that `gapdoc` does not like these.

**Comment by simpcomp-team** [comment]
In `release-gap-packages`, the file `makedoc.g` is called twice. Once from 

`/home/jonathan/bin/gap-4.11.1/pkg/[package]`

on line 274, and once from

`/home/jonathan/bin/gap-4.11.1/pkg/[package]/tmp/[package]-[version]`

on line 424. 

The first command works because it finds my files with the pieces for the manual (in `doc/gapdoc`). In the second command, those files are not found (and hence the list of files passed to `ComposedDocument` in ComposeXML.gi in the GAPDoc folder is empty).

So, is it possible that line 423 in  `release-gap-packages`:

`PushOptions(rec(relativePath:="../../.."));`

causes those files in `doc/gapdoc` not to be found?

**Comment by fingolfin** [comment]
No, that command just ensures relative links in the generated HTML files are set up in a way that makes them work in a regular GAP installation.

what is the exact error message you are seeing?

just to ensure we are on the same page: are you using the latest version of the release script?

which version of GAPDoc is installed?

**Comment by fingolfin** [comment]
So I've used a modified version of `release-gap-package` (where I disabled the check for the tag on GitHub, as (a) you tagged a slightly older commit already, and should delete that tag/release first; and (b) of course I don't have write access to your repo).

I also made sure to use it with stock GAP 4.11.1, with the default set of packages bundled with that (so e.g. GAPDoc 1.6.4). And it worked just fine, it managed to compile the manuals twice.

**Comment by fingolfin** [comment]
Also, I would strongly suggest to update the release date in `PackageInfo.g` to today; releasing a package today with release date set to `07/06/2021` is misleading

**Comment by fingolfin** [comment]
There are more problems with the manual code, I'll post another PR shortly

**Comment by simpcomp-team** [comment]
I am using the latest version of the release script (downloaded it yesterday after a number of failed attempts with an older version of the script), GAPDoc version is 1.6.4.

My package info file shows a release date of yesterday. I had not pushed it to github, this is done now. 

The error message I am seeing is this one:

`Generating documentation in Directory("./doc/")
#I Composing XML document . . .
Error, Did not find chunk SCObjects at /home/jonathan/bin/gap-4.11.1/pkg/GAPDoc-1.6.4/lib/ComposeXML.gi:233 called from`

I can explain what is happening: In the first run through the script finds all the right files in the right directory. In the second run through the list of files which is used to collect the chunks for the documentation is empty. So: the record `pieces` is empty. And the first key the script is looking for is `SCObjects`.

Everything compiles if I add `lib` to `scan_dirs` in `makedoc.g`, but then it simply uses the files from `lib` and they don't contain the correct chunks for the manual.

As for the tags: a tag is created every time I run the script past the first built of the manual. If I want to run the script again it does not let me without deleting the tag first. Not sure how best to roll back everything that has happened after the failed run of the release script.

**Comment by simpcomp-team** [comment]
I agree, with `makedoc.g` like

`AutoDoc("simpcomp",rec(gapdoc:=rec(scan_dirs:=["doc/gapdoc","lib"])));`

(this is the version you used) it works and this is what I used for the release of 2.1.13, but the examples in the manual are now outdated.

**Comment by simpcomp-team** [comment]
Sorry about that. I will make sure to have the github remote up to date when I try this in the future. I just did not want half an update in github in case the release script does not run through.

**Comment by fingolfin** [comment]
Just to be clear, that wasn't meant as an accusation, no need to be sorry, it happens. And it wasn't hard to rebase -- all PRs should be ready now. And PR #33 might resolve your problems.

**Comment by fingolfin** [comment]
They are actually *defined* as part of that PR -- inside `PackageInfo.g`:
```
AutoDoc := rec(
    # provide some entities to be used on the title page
    entities := rec(
        VERSION := ~.Version,
        DATE := ~.Date,
        YEAR := SplitString(~.Date,"/")[3],
        ),
    TitlePage := rec( ), # Empty TitlePage here is a workaround for an AutoDoc bug
),
```

That's a feature of AutoDoc.

BTW, PRs #32 and #4 remove two more places that have the version. So when you merge those, you only need to update the version and release date in `PackageInfo.g` (and perhaps `CHANGELOG`).

**Comment by simpcomp-team** [comment]
Ah, right, I see.

**Comment by fingolfin** [comment]
Please let's wait for PR #37. It would be good to first have all CI tests pass, including those building the manual.

**Comment by simpcomp-team** [comment]
I think the manual built in #37 failed because there was an unresolved conflict in howto.xml.

I updated this a minute ago before I saw your message.

**Comment by fingolfin** [comment]
The output for `SCRunTest` you now have in the manual does not match what it actually outputs. BTW that whole function essentially duplicates what `TestPackage("simpcomp")` does :-)

**Comment by simpcomp-team** [comment]
Yes, calling it `SCRunTest` was part of Felix Effenbergers effort to brand the package a long time ago. It suppresses the output about version number and test file used.

**Comment by simpcomp-team** [comment]
Should I try and merge now? It seems the manual builds now, but the other tests failed.

**Comment by fingolfin** [comment]
Maybe close issue #21, then try again to make a release?

**Comment by fingolfin** [comment]
@simpcomp-team almost good now... except #41 :-/ but as explained there, no need for a new version etc. Might also want to merge #40 while at it, but unimportant

**Comment by fingolfin** [comment]
OK, now still needs `release-gap-package --force`

**Comment by simpcomp-team** [comment]
Before I do this, I would like to revisit the issue with the manual: if permitted, the manual still uses the files from `lib` as a resource. But I have a script that rewrites the examples and saves them into another folder `doc/gapdoc`. This second folder is not found in the second run of `makedoc.g` in the release script. I have no idea why because everything is fine for the first run (if I run `makedoc.g` locally then everything is fine as well). If `lib` is added to `scan_dirs` in `makedoc.g` then it simply takes the files from there and they don't contain the correct comments for the manual.

**Comment by fingolfin** [comment]
I sent you an email, I think it'd be best to talk about this on a video chat or phone, else we'll just comment past each other :-).

**Comment by fingolfin** [comment]
And this new code in `makedoc.g` is definitely wrong, as it will pick the wrong files in general: `[Concatenation(GAPInfo.RootPaths[2],"pkg/simpcomp/doc/gapdoc")]`

This won't even work in my setup, because `simpcomp` is installed in a completely different location.

Anyway: there is no `doc/gapdoc/` directory in the repository, nor in the release tarball, so this can't work. Perhaps you have one locally, that you don't want to commit? Then you must make sure to (re-)create it during the creation of the release tarball. E.g. via a `.release` script

**Comment by fingolfin** [comment]
To expand: you can add an executable file `.release` to your git repository. If found, `release-gap-package` executes this script during the release process, inside the `tmp/simpcomp-X.Y` dir (the `.release` script itself won't be included in the release).

Other packages also use this script to remove files that are not supposed to be distributed, e.g. you could add `rm -rf internal` to it.

(As to my video chat offer: it stands in principle, but I am going to get some lunch now, will be back later, if you are interested, feel free to suggest a time and we can see if we can make it happen :-) )

**Comment by fingolfin** [comment]
Hrm, there are also issue with https://github.com/simpcomp-team/simpcomp/releases/download/v2.1.13/simpcomp-2.1.13.tar.gz -- e.g. it does not look as if `autogen.sh` did run (which `release-gap-package` normally does automatically), rendering the tarball unusable. I wonder what went wrong there? Definitely one more thing to watch out for.

**Comment by simpcomp-team** [comment]
So, you are saying that `doc/gapdoc` must be present in `tmp/simpcomp-X.Y`? So, could I just add

`cp -r ../../doc/gapdoc doc`

to `.release` and that could be it?

**Comment by simpcomp-team** [comment]
So, the first run of `makedoc.g` happens in the repository, the second run happens in `tmp/simpcomp-X.Y`? This explains a lot. 

Now the release script produced something, but there are still some errors (and .release is probably not what you imagine it should be and I had to change the release script - so all not ideal).

Here is the output from the release script:

`Extracting information from PackageInfo.g...
Package simpcomp 2.1.14
Building GAP package documentation (using makedoc.g)
Using archive basename simpcomp-2.1.14
Using GitHub repository simpcomp-team/simpcomp
Using GitHub username simpcomp-team
Using branch master
Creating git tag v2.1.14

Exporting repository content for tag 'v2.1.14'
Building GAP package documentation for archives (using makedoc.g)
Removing unnecessary files
Generating build system files
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force ${ACLOCAL_FLAGS}
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
autoreconf: Leaving directory `.'

Validating PackageInfo.g...

Creating simpcomp-2.1.14.tar.gz ...
Pushing your branch to GitHub
Everything up-to-date
Pushing your tag to GitHub
Total 0 (delta 0), reused 0 (delta 0)
To https://github.com/simpcomp-team/simpcomp
 + 3ba1e6be...f509dddc v2.1.14 -> v2.1.14 (forced update)
Creating new release v2.1.14 on GitHub

Uploading simpcomp-2.1.14.tar.gz with mime type application/x-gzip
######################################################################### 100.0%

../../ReleaseTools/release-gap-package: 634: ../../ReleaseTools/release-gap-package: Bad substitution`

**Comment by simpcomp-team** [comment]
Now the release is broken / incomplete. The manual in the tmp directory is now the right one, but it is not uploaded.

**Comment by fingolfin** [comment]
Somehow something is still very wrong. (BTW I am now again in the Jitsi channel to which I sent you a link on Friday; but I'll be on and off, as today our regular child care does not happen and I have to look after both kids)

Anyway, let's see:
- the archive is still missing various files generated by `./autogen.sh`, such as `install-sh` and `missing` -- and thus is unusable by regular users. This is quite concerning, and I am flabbergasted as to what might be wrong here, as it so far has worked for dozens and dozens of packages and different package authors, and I right now have no clue at all what might possibly go wrong there
- yes, there are two runs of `makedoc.g` in two different directories, as you gathered. 
- you said you had to edit the release script; what exactly did you change? It is difficult to debug this if we look at different versions of the code :-)
- I tweaked your changes to `.release` a bit, see PR #43; copying from the original dir sounds OK...
- You write: "The manual in the tmp directory is now the right one, but it is not uploaded" -- the manual of course is there, inside the simpcomp tarball. But I assume that no manual.pdf was uploaded. That's likely related to this error: `../../ReleaseTools/release-gap-package: 634: ../../ReleaseTools/release-gap-package: Bad substitution`

**Comment by fingolfin** [comment]
Here is what I get with the very latest version of `release-gap-package` (just pushed; only relevant changes are a fix for `--only-tarball`); note that this skips uploading anything. My main goal is to verify the content of the resulting tarball
```
$ release-gap-package --only-tarball
Extracting information from PackageInfo.g...
Package simpcomp 2.1.14
Building GAP package documentation (using makedoc.g)
Using archive basename simpcomp-2.1.14
Using GitHub repository simpcomp-team/simpcomp
Using GitHub username fingolfin
Using branch mh/release
Using git tag v2.1.14 (derived from ArchiveURL in in PackageInfo.g)
WARNING: release v2.1.14 already exists on GitHub, but only tarballs were requested, so continue anyway
Exporting repository content for tag 'v2.1.14'
Building GAP package documentation for archives (using makedoc.g)
Removing unnecessary files
Generating build system files
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force ${ACLOCAL_FLAGS}
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /opt/homebrew/Cellar/autoconf/2.71/bin/autoconf --force
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
autoreconf: './install-sh' is created
autoreconf: Leaving directory '.'

Validating PackageInfo.g...

Creating simpcomp-2.1.14.tar.gz ...
$ ls -F tmp/simpcomp-2.1.14
AUTHORS    Makefile.am    README.md    complexes/    doc/         lib/       src/
CHANGELOG  Makefile.in    aclocal.m4   configure*    init.g       makedoc.g  tst/
COPYING    PackageInfo.g  autogen.sh*  configure.ac  install-sh*  read.g
$
```

As you can see, `install-sh` is present, and indeed, running `./configure` in that dir works.

I do have a slightly newer versions of autoconf (2.71 versus 2.69) and automake (1.16.5 versus 1.15.1) than you, but that shouldn't make a difference...

**Comment by fingolfin** [comment]
The `Bad substitution` was a recent regression in `release-gap-packages`, which should be fixed now.

**Comment by simpcomp-team** [comment]
The data in `doc/gapdoc` contains the code for the examples in the manual that can automatically be re-generated if necessary. In theory, the lib files should only contain the commands of the examples. In practice, it sometimes still contains output of earlier simpcomp versions. I should clean this up at some point. 

This setup proved to be very convenient for v2.1.14 because a simplicial complex is now printed differently. It sometimes also picks up errors that `TestPackage("simpcomp");` misses. 

I think the script to run the examples is nothing that should be included in the repository. I run it sometimes, but not for every minor update. It is really more of an internal tool. In the long run it would probably make sense to include it into `.release`.

**Comment by simpcomp-team** [comment]
About my changes to the release script: I copied the run of `.release` to line 429. I am not sure if I use this right, but I copy .release to `tmp/simpcomp-X.Y,` then run it in `tmp/simpcomp-X.Y`.

As for the missing files in the repository: you are talking about all the files generated by running `autogen.sh`? Locally all this seems to work.

**Comment by fingolfin** [comment]
> About my changes to the release script: I copied the run of `.release` to line 429. 

OK...

> I am not sure if I use this right, but I copy .release to `tmp/simpcomp-X.Y,` then run it in `tmp/simpcomp-X.Y`.

This is extremely strange. You should not have to copy .release, it should be there. If it is not, something is very, very wrong.

> 
> As for the missing files in the repository: you are talking about all the files generated by running `autogen.sh`?

Exactly. They are not in the archive.

> Locally all this seems to work.

What do you mean by that? Are they present in `tmp/simpcomp-X.Y` after release-gap-package finishes?

**Comment by fingolfin** [comment]
I've just downloaded the simpcomp 2.1.14 tarball you produced now, and it seems to work for me!! Great!

So I'll now test it some more, and assuming all tests pass, we hopefully can close this issue.

Thank you very much for sticking through this tiresome process and not throwing the towel long before

**Comment by fingolfin** [comment]
OK, not sure why the tarball worked for me earlier, but unfortunately, in CI tests using your current 2.1.14 tarball, I am getting this:
```
configure: error: cannot find install-sh, install.sh, or shtool in "." "./.." "./../.."
```
So yeah, it still is missing those files. Indeed:
```
$ ls simpcomp-2.1.14
AUTHORS    Makefile.am    README.md   complexes     doc     makedoc.g  tst
CHANGELOG  Makefile.in    aclocal.m4  configure     init.g  read.g
COPYING    PackageInfo.g  autogen.sh  configure.ac  lib     src
```

Can you please verify the following:
- in your `simpcomp` git clone directory, do `rm -f install-sh missing && ./autogen.sh` -- are the files `install-sh` and `missing` regenerated?
- after running `release-gap-packages`, it leaves behind the `tmp/simpcomp-2.1.14` directory; does this contain `install-sh` and `missing` ?

**Comment by simpcomp-team** [comment]
Mhh, too bad.

As for your request:

- `install-sh` and `missing` are regenerated by (first deleting them and then) running `autogen.sh`
- `tmp/simpcomp-2.1.14` does not contain these files

**Comment by fingolfin** [comment]
if you manually run `./autogen.sh` in `tmp/simpcomp-2.1.14`, are the files added?

If so, I wonder why `release-gap-package` does not add them?!?

**Comment by fingolfin** [comment]
It is especially baffling because `configure` is also created by `autogen.sh` and that file is of course present in the tarball.

**Comment by fingolfin** [comment]
Also, it does work for me. Which makes me wonder if the versions of autoconf/automake involved are somehow at fault after all?

Anyway: I guess one can try to hack around it -- see PR #45

**Comment by simpcomp-team** [comment]
Here is what I can say:

Running `autogen.sh` in `tmp/simpcomp-2.1.12` produces `missing` and `install.sh.`
Running `autogen.sh` in `tmp/simpcomp-2.1.13` and `14` does not produce `missing` and `install.sh.`

This must be some incredibly stupid error somewhere.

**Comment by simpcomp-team** [comment]
I merged PR #45 and released again. Hope this works now. But we should keep an eye on this. 

If this release works I will close this issue and create a new one about the `autoreconf` issue so I don't forget.

**Comment by fingolfin** [comment]
It worked, thanks!

However, tests fail if `grape` is not loaded... perhaps you'd be willing to merge #46 and make one last iteration of the release?

**Comment by simpcomp-team** [comment]
Done.  Thank you so much for all your help with this.

**Comment by fingolfin** [comment]
The release has now been picked up in the "new package distribution". It'll be in GAP 4.12.
